src/player_display.cpp

```diff
@@ -219,13 +219,165 @@ std::string get_encumbrance_description( const player &p, body_part bp, bool com
+    void draw_active_stats(player& player, input_context &ctxt);
+    void draw_active_encumbrance(player& player, input_context &ctxt);
+    void draw_active_traits(player& player, input_context &ctxt);
+    void draw_active_bionics(player& player, input_context &ctxt);
+    void draw_active_effects(input_context &ctxt);
+    void draw_active_skills(player& player, input_context &ctxt);
+
+    void draw_inactive_stats();
+    void draw_inactive_encumbrance();
+    void draw_inactive_traits(player& player);
+    void draw_inactive_bionics(player& player);
+    void draw_inactive_effects();
+    void draw_inactive_skills(player& player);
+
+private:
+    enum tab : int {
+        STATS = 1,
+        ENCUMBRANCE = 2,
+        SKILLS = 3,
+        TRAITS = 4,
+        BIONICS = 5,
+        EFFECTS = 6
+    };
+
+    void init_effect_window(player& player);
+    void init_bionics_window();
+    void increment_tab();
+    void decrement_tab();
+
+    bool done;
+    tab current_tab;
     unsigned line;
+
+    catacurses::window w_grid_top;
+    catacurses::window w_grid_skill;
+    catacurses::window w_grid_trait;
+    catacurses::window w_grid_bionics;
+    catacurses::window w_grid_effect;
+
+    catacurses::window w_tip;
+    catacurses::window w_stats;
+    catacurses::window w_traits;
+    catacurses::window w_bionics;
+    catacurses::window w_encumb;
+    catacurses::window w_effects;
+    catacurses::window w_speed;
+    catacurses::window w_skills;
+    catacurses::window w_info;
+
+    unsigned effect_win_size_y;
+    unsigned trait_win_size_y;
+    unsigned bionics_win_size_y;
+    unsigned skill_win_size_y;
+    unsigned info_win_size_y;
+
+    unsigned infooffsetytop;
+    unsigned infooffsetybottom;
+
     std::vector<std::string> effect_name;
     std::vector<std::string> effect_text;
+
+    std::vector<Skill const *> skillslist;
+
+    const std::string title_STATS;
+    const std::string title_ENCUMB;
+    const std::string title_TRAITS;
+    const std::string title_BIONICS;
+    const std::string title_EFFECTS;
+    const std::string title_SKILLS;
+    const std::string title_SPEED;
+};
+
+void player_display_info::increment_tab()
+{
+    current_tab = (current_tab >= EFFECTS)
+        ? STATS
+        : static_cast<tab>(static_cast<int>(current_tab)+1);
+}
+void player_display_info::decrement_tab()
+{
+    current_tab = (current_tab <= STATS)
+        ? EFFECTS
+        : static_cast<tab>(static_cast<int>(current_tab)-1);
+}
+
+player_display_info::player_display_info(player& player) :
+    done(false),
+    current_tab(STATS),
+    line(0),
+    title_STATS(_( "STATS" )),
+    title_ENCUMB(_( "ENCUMBRANCE AND WARMTH" )),
+    title_TRAITS(_( "TRAITS" )),
+    title_BIONICS(_( "BIONICS" )),
+    title_EFFECTS(_( "EFFECTS" )),
+    title_SKILLS(_( "SKILLS" )),
+    title_SPEED(_( "SPEED" ))
+{
+    this->init_effect_window(player);
+    effect_win_size_y = 1 + unsigned( effect_name.size() );
```

Comment:
`size()` yields a `size_t`, which is already unsigned, but not necessarily an `int`. Use `size_t` instead of `unsigned` for consistency and you don't need to cast this.